R/car.R

```diff
@@ -74,7 +74,17 @@ scale_gmrf_precision <- function(Q,
                                  A = matrix(1, ncol = ncol(Q)),
                                  eps = sqrt(.Machine$double.eps)) {
 
-  nb <- spdep::mat2listw(abs(Q), style = "M")$neighbours
+  # spdep is annoying warning with style = "M" but not clear from docs
+  # if this means style missing or style Matrix. And none of the other
+  # options for style seemingly do the same thing as style = "M"
+  # so just ignoring this warning for now
+  withCallingHandlers({
+    nb <- spdep::mat2listw(abs(Q), style = "M")$neighbours
```

Comment:
There are new warnings coming from this.

In the current build

```
Warning (test-car.R:12:3): scale_gmrf_precision() returns expected values
6 is not an origin
Backtrace:
    ▆
 1. └─naomi::scale_gmrf_precision(Q) at test-car.R:12:3
 2.   ├─base::withCallingHandlers(...) at src/R/car.R:81:3
 3.   └─spdep::mat2listw(abs(Q), style = "M") at src/R/car.R:82:5
 4.     └─spdep::sn2listw(...)
Warning (test-car.R:12:3): scale_gmrf_precision() returns expected values
no-neighbour observations found, set zero.policy to TRUE;
this warning will soon become an error
Backtrace:
    ▆
 1. └─naomi::scale_gmrf_precision(Q) at test-car.R:12:3
 2.   ├─base::withCallingHandlers(...) at src/R/car.R:81:3
 3.   └─spdep::mat2listw(abs(Q), style = "M") at src/R/car.R:82:5
 4.     └─spdep::sn2listw(...)
```

Is there anything we can do about this? Can we replace this function entirely? It feels like this package is always chucking out warnings and there is a new one every few weeks.

Reply:
@jeffeaton I thank we have discussed this in the past - any thoughts?

Reply:
just confirming this is resolved now in #418.